### ISSUE TITLE
Code does not follow the documented default values

### ISSUE THREAD

**Issue by yvesf** [grails-plugins/grails-quartz#47]
Grails 3.0.11 / org.grails.plugin:quartz:2.0.4

In QuartzGrailsPlugin:
```groovy
        boolean pluginEnabled = grailsApplication?.config?.quartz?.pluginEnabled
        boolean autoStart = grailsApplication?.config?.quartz?.autoStartup
```

If value is not set then this results in false (empty set -> false). This contradicts the documented defaults of these two properties.

Once I set both options explicitly to true it will fail again:
In line 89:
```
       if (grailsApplication?.config?.quartz?.purgeQuartzTablesOnStartup?.toBoolean()==true) {
```

The config object apparently does no longer return null if a property is not set. Instead a empty Map is returned. There is no toBoolean() on that. I propose to remove the toBoolean() call. This way it will default to false because `[:] == true` equals false.

To sum up, I had to set the following options although there should be sane default values:
```
quartz:
    autoStartup: true
    pluginEnabled: true
    purgeQuartzTablesOnStartup: false
```

**Comment by yvesf** [comment]
@rvanderwerf I just saw your commit. Thanks for the very fast response!

I just had a look on the default values issue, to me it looks like we should write it like this instead of checking for NullSafeNavigator
```groovy
def pluginEnabled = grailsApplication.config.getProperty('quartz.pluginEnabled', Boolean, true)
def autoStartup = grailsApplication.config.getProperty('quartz.autoStartup', Boolean, true)
```

**Comment by graemerocher** [comment]
@yvesf yes you are correct, you should avoid navigating the config via the dot operator as it is slower

**Comment by rvanderwerf** [comment]
PR welcome on the checking, but the issue raised is already fixed on 2.0.5. If it's not working for you feel free to re-open.
